crates/aptos-ledger/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+    // open connection to ledger
+    let transport = match open_ledger_transport() {
+        Ok(transport) => transport,
+        Err(err) => return Err(err),
+    };
+
+    match transport.exchange(&APDUCommand {
+        cla: CLA_APTOS,
+        ins: INS_GET_VERSION,
+        p1: P1_NON_CONFIRM,
+        p2: P2_LAST,
+        data: vec![],
+    }) {
+        Ok(response) => {
+            // Received response from Ledger
+            if response.retcode() == APDU_CODE_SUCCESS {
+                let major = response.data()[0];
+                let minor = response.data()[1];
+                let patch = response.data()[2];
+                let version = format!("{}.{}.{}", major, minor, patch);
+                Ok(version)
+            } else {
+                let error_string = response
+                    .error_code()
+                    .map(|error_code| error_code.to_string())
+                    .unwrap_or_else(|retcode| format!("Error with retcode: {:x}", retcode));
+                Err(AptosLedgerError::UnexpectedError(error_string))
+            }
+        },
+        Err(err) => Err(AptosLedgerError::UnexpectedError(err.to_string())),
+    }
+}
+
+/// Returns the official app name register in Ledger
+pub fn get_app_name() -> Result<String, AptosLedgerError> {
+    // open connection to ledger
+    let transport = match open_ledger_transport() {
+        Ok(transport) => transport,
+        Err(err) => return Err(err),
+    };
+
+    match transport.exchange(&APDUCommand {
+        cla: CLA_APTOS,
+        ins: INS_GET_APP_NAME,
+        p1: P1_NON_CONFIRM,
+        p2: P2_LAST,
+        data: vec![],
+    }) {
+        Ok(response) => {
+            if response.retcode() == APDU_CODE_SUCCESS {
+                let app_name = match str::from_utf8(response.data()) {
+                    Ok(v) => v,
+                    Err(e) => return Err(AptosLedgerError::UnexpectedError(e.to_string())),
+                };
+                Ok(app_name.to_string())
+            } else {
+                let error_string = response
+                    .error_code()
+                    .map(|error_code| error_code.to_string())
+                    .unwrap_or_else(|retcode| format!("Error with retcode: {:x}", retcode));
+                Err(AptosLedgerError::UnexpectedError(error_string))
+            }
+        },
+        Err(err) => Err(AptosLedgerError::UnexpectedError(err.to_string())),
+    }
+}
+
+/// Returns the public key of your Aptos account in Ledger device
+///
+/// # Arguments
+///
+/// * `display` - If true, the public key will be displayed on the Ledger device, and confirmation is needed
+pub fn get_public_key(display: bool) -> Result<String, AptosLedgerError> {
+    // open connection to ledger
+    let transport = match open_ledger_transport() {
+        Ok(transport) => transport,
+        Err(err) => return Err(err),
+    };
+
+    // serialize the derivative path
+    let cdata = serialize_bip32(DERIVATIVE_PATH);
```

Comment:
It seems like this is the same every time, let's use this: https://docs.rs/once_cell/latest/once_cell/sync/struct.Lazy.html.

Reply:
Yes for now. In the future, we will need to support multiple index account, and the derivative_path will be different depending on the account/address index. (Current version only support Account created at index 0)

Reply:
We should still use Lazy for this and each of those derivative paths if we know them ahead of time.

Reply:
Yep, added Lazy for now. I will follow up with a new design on how to handle the multiple index accounts, since there could be a lot of them (10+)